### DISCUSSION TITLE
Sub application under main site

### DISCUSSION THREAD

**Discussion by daxydoggie** [x-govuk/govuk-frontend-aspnetcore#278] [Q&A] [answered]
hi, i am using this component on a razor .net6 site, i have to deploy it as a sub application/site, as we need to share the 80/443 bindings/certs with many other sites under the same host. no matter how i set the options in program.cs for the component using StaticAssetsContentPath and CompiledContentPath, it seems it is always resolved at runtime relative to the host root and not the application root as there seems to be validation/bootstrapping that requires the properties to start with a "/" (not "~ /" or even no slash)
Can you shed some light as to how i may do this relative to the app path and not the host path if not being able to use "~ /" or simply just "foldername", ie something relative to the app not the host.

I did think at first trying "/mysitename/govukassets", but this just kicks the can down the road as only "/mysitename/mysitename/goveukassets" can be resolved. I can get it working somewhat by manually creating the folders and copying the css/js/fonts but kinda defeats the purpose of the component nicely loading everything for me.

Just to confirm I am running the site as an application under the main site.

https://some-local-council.gov.uk/mysitename

thanks for a great product btw, just been frustrating since having to deploy as sub app on iis.

**Comment by gunndabad** [comment]
I suspect this is due a missing `Request.PathBase` somewhere where the links are generated. I’ll get a fix pushed.

**Reply by daxydoggie** [reply]
thats great thanks for that, i'll keep an eye on the repo for the fix.

**Reply by gunndabad** [reply, marked as answer]
@daxydoggie could you try out https://www.nuget.org/packages/GovUk.Frontend.AspNetCore/2.1.0-rc and see if it works for you? It should _just work_ without doing anything special with `StaticAssetsContentPath` or `CompiledContentPath`.

**Reply by daxydoggie** [reply]
> @daxydoggie could you try out https://www.nuget.org/packages/GovUk.Frontend.AspNetCore/2.1.0-rc and see if it works for you? It should _just work_ without doing anything special with `StaticAssetsContentPath` or `CompiledContentPath`.

That's great James, works like a charm. 
When is the next release cycle for 2.1?

Thanks, Craig.

**Reply by gunndabad** [reply]
That's good to hear. I'll add some docs around this then release 2.1. It'll probably be towards the end of this week.
